lib/ClangImporter/SwiftLookupTable.cpp

```diff
@@ -1157,27 +1179,50 @@ void SwiftLookupTableWriter::writeExtensionContents(
 namespace {
   /// Used to deserialize the on-disk base name -> entities table.
   class BaseNameToEntitiesTableReaderInfo {
+    ASTContext &ctx;
+
   public:
-    using internal_key_type = StringRef;
-    using external_key_type = internal_key_type;
+    using internal_key_type = std::pair<DeclNameKind, StringRef>;
+    using external_key_type = DeclName;
     using data_type = SmallVector<SwiftLookupTable::FullTableEntry, 2>;
     using hash_value_type = uint32_t;
     using offset_type = unsigned;
 
+    BaseNameToEntitiesTableReaderInfo(ASTContext &context) : ctx(context) {}
+
     internal_key_type GetInternalKey(external_key_type key) {
-      return key;
+      return {key.getKind(), key.str()};
     }
 
     external_key_type GetExternalKey(internal_key_type key) {
-      return key;
+      switch (key.first) {
+      case swift::DeclNameKind::Normal:
+        return DeclName(ctx.getIdentifier(key.second));
```

Comment:
Ah, I didn't think about a growing identifier table. Now it makes sense that `StringRef` was used throughout that part. I'll try and change some stuff back to strings then.